chain/vm/vm.go

```diff
@@ -602,36 +602,36 @@ func (vm *VM) incrementNonce(addr address.Address) error {
 	})
 }
 
-func (vm *VM) transfer(from, to address.Address, amt types.BigInt) error {
+func (vm *VM) transfer(from, to address.Address, amt types.BigInt) aerrors.ActorError {
 	if from == to {
 		return nil
 	}
 
 	if amt.LessThan(types.NewInt(0)) {
-		return xerrors.Errorf("attempted to transfer negative value")
+		return aerrors.New(exitcode.SysErrSenderInvalid, "attempted to transfer negative value")
 	}
 
 	f, err := vm.cstate.GetActor(from)
 	if err != nil {
-		return xerrors.Errorf("transfer failed when retrieving sender actor")
+		return aerrors.Newf(exitcode.SysErrSenderInvalid, "transfer failed when retrieving sender actor: %s", err)
 	}
 
 	t, err := vm.cstate.GetActor(to)
 	if err != nil {
-		return xerrors.Errorf("transfer failed when retrieving receiver actor")
+		return aerrors.Newf(exitcode.SysErrInvalidReceiver, "transfer failed when retrieving receiver actor: %s", err)
 	}
 
 	if err := deductFunds(f, amt); err != nil {
-		return err
+		return aerrors.Newf(exitcode.SysErrInsufficientFunds, "transfer failed when deducting funds: %s", err)
 	}
 	depositFunds(t, amt)
 
 	if err := vm.cstate.SetActor(from, f); err != nil {
-		return err
+		return aerrors.Newf(exitcode.ErrIllegalState, "transfer failed when setting receiver actor: %s", err)
```

Comment:
Both this and the one below must be fatal errors in the VM. This is not a condition repeatable in other impls, it's probably a storage problem or something.

Also, the VM must never use exit code that don't begin with `Sys`. There is no `SysErrIllegalState` because that means fatal error that should not go on chain.